File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.java
class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        int sum;
        for (int i = 0; i < photo.length; i++) {
            sum = 0;
            for (int j = 0; j < name.length; j++) {
                for (int k = 0; k < photo[i].length; k++) {
                    if (photo[i][k].equals(name[j])) {
                        sum += yearning[j];
                    }
                }
            }
            answer[i] = sum;
        }
        return answer;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    for i in range(len(photo)):
        lst = photo[i]
        res = 0
        for j in lst:
            if j in name:
                res += yearning[name.index(j)]
        answer.append(res)
    return answer